### PR TITLE
chore: configure secrets for deployment

### PR DESCRIPTION
**Pull request by algomaster99**
The changes configure the passwords required for deployment.
- Spoonbot private key
- Spoon-bot's Sonatype's credentials

@monperrus Could you give me your GPG ID (I looked it up online and it says that the message does not include the public key of the sender) so that I could send you the [passwords you need to configure for this repository](https://docs.github.com/en/actions/security-guides/encrypted-secrets#creating-encrypted-secrets-for-a-repository)?

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -25,4 +25,8 @@ jobs:
       with:
         java-version: 11
     - name: Start deployment
+      env:
+        CI_DEPLOY_USERNAME: ${{ secrets.CI_DEPLOY_USERNAME }}
+        CI_DEPLOY_PASSWORD: ${{ secrets.CI_DEPLOY_PASSWORD }}
+        SPOONBOT_PARAPHRASE: ${{ secrets.SPOONBOT_PARAPHRASE }}
```
Comment (algomaster99) [inline .github/workflows/publish.yml:31]: These are the secrets I will send you via the encrypted channel and then we shall try if we are able to deploy.

File: .buildscript/settings.xml [hunk 2]
```diff
@@ -1,14 +1,9 @@
...
-        <username>${env.CI_DEPLOY_USERNAME}</username>
-        <password>${env.CI_DEPLOY_PASSWORD}</password>
-    </server>
-</servers>
+      <server>
+          <id>ossrh</id>
+          <username>${env.CI_DEPLOY_USERNAME}</username>
+          <password>${env.CI_DEPLOY_PASSWORD}</password>
+      </server>
+  </servers>
```
Comment (algomaster99) [inline .buildscript/settings.xml:8]: You sent me credentials for `ossrh` only and not `ossrh-snapshot` so I deleted the XML for the latter.

Comment (monperrus) [reply .buildscript/settings.xml:8]: the identifier is arbitrary, it is only a relation between the pom.xml and settings.xml.

what matters is to have a proper Sonatype token

File: .github/workflows/publish.yml (hunk 1 above)
Comment (monperrus) [reply .github/workflows/publish.yml:31]: ack

File: .buildscript/settings.xml (hunk 2 above)
Comment (algomaster99) [reply .buildscript/settings.xml:8]: OK. If it's arbitrary, I will revert this change.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (algomaster99) [reply .github/workflows/publish.yml:31]: @monperrus Could you send me your GPG ID for that?

**Comment by monperrus** [comment]
When you received my encrypted message, the signature tells you the GPG key I use.

It's https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x074f73b36d8dd649b132bac18035014a2b7bfa92

**Comment by algomaster99** [comment]
@monperrus I checked online how I could retrieve the sender's public key using a PGP message but I found this [post](https://crypto.stackexchange.com/a/81919) which says that it is not possible to get the public key of the sender using the PGP message. Hence, I asked you for your key.

**Comment by algomaster99** [comment]
@monperrus Once you configure the secrets, please merge this PR. I hope it will work :crossed_fingers: .

**Comment by monperrus** [comment]
Thanks @algomaster99

**Comment by algomaster99** [comment]
@monperrus The deployment failed again. Let me try from local now since I have the paraphrase. 

On the bright side, we were able to import spoon-bot key. What is it for, by the way?

**Comment by monperrus** [comment]
> On the bright side, we were able to import spoon-bot key. What is it for, by the way?

To upload a release to Maven Central, it is mandatory to GPG-sign it.

**Comment by algomaster99** [comment]
@monperrus the spoon bot key expired on 25th July 2019. I am unsure how it was working before but I can't use it to deploy.

**Comment by algomaster99** [comment]
I updated the spoon-bot key expiration date and submitted the [PR](https://github.com/SpoonLabs/gumtree-spoon-ast-diff/pull/209) for it.

**Pull request merged**